Invalid escape sequence \. in kokkos_functions.cmake

https://github.com/kokkos/kokkos/blob/d3a941925cbfb71785d8ea68259123ed52d3f9da/cmake/kokkos_functions.cmake#L50

All:

I was building Trilinos using an ancient version of Cmake (2.8.12.2) and ran into this particular error:

CMake Error at packages/kokkos/cmake/kokkos_functions.cmake:50 (string):
  Syntax error in cmake code at

    /packages/Trilinos/packages/kokkos/cmake/kokkos_functions.cmake:50

  when parsing string

    [0-9]+\.[0-9]+\.[0-9]+$

  Invalid escape sequence \.


I was able to fix this issue by changing the REGEX to [0-9]+\\.[0-9]+\\.[0-9]+$ . Should this be changed? It may be a change in how REGEX is handled in newer versions of Cmake.

Thanks!

@mhoemmen

--- a/cmake/kokkos_functions.cmake
+++ b/cmake/kokkos_functions.cmake
@@ -47,7 +47,7 @@ function(set_kokkos_cxx_compiler)
                     OUTPUT_VARIABLE INTERNAL_CXX_COMPILER_VERSION
                     OUTPUT_STRIP_TRAILING_WHITESPACE)
 
-    string(REGEX MATCH "[0-9]+\.[0-9]+\.[0-9]+$"
+    string(REGEX MATCH "[0-9]+\\.[0-9]+\\.[0-9]+$"
            INTERNAL_CXX_COMPILER_VERSION ${INTERNAL_CXX_COMPILER_VERSION})
   endif()
 

